nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateFakeRecord.java

```diff
@@ -0,0 +1,1021 @@
+            .build();
+
+    static final PropertyDescriptor LOCALE = new PropertyDescriptor.Builder()
+            .name("gen-fake-record-locale")
+            .displayName("Locale")
+            .description("The locale that will be used to generate field data. For example a Locale of 'es' will generate fields (e.g. names) in Spanish.")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .defaultValue("en-US")
+            .allowableValues(SUPPORTED_LOCALES)
+            .build();
+
+    static final PropertyDescriptor NULLABLE_FIELDS = new PropertyDescriptor.Builder()
+            .name("gen-fake-record-nullable-fields")
+            .displayName("Nullable Fields")
+            .description("Whether the generated fields will be nullable. Note that this property is ignored if Schema Text is set. Also it only affects the schema of the generated data, " +
+                    "not whether any values will be null. If this property is true, see 'Null Value Percentage' to set the probability that any generated field will be null.")
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .required(true)
+            .build();
+    static final PropertyDescriptor NULL_PERCENTAGE = new PropertyDescriptor.Builder()
+            .name("gen-fake-record-null-pct")
+            .displayName("Null Value Percentage")
+            .description("The percent probability (0-100%) that a generated value for any nullable field will be null. Set this property to zero to have no null values, or 100 to have all " +
+                    "null values.")
+            .addValidator(StandardValidators.createLongValidator(0L, 100L, true))
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .required(true)
+            .defaultValue("0")
+            .dependsOn(NULLABLE_FIELDS, "true")
+            .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("FlowFiles that are successfully transformed will be routed to this relationship")
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(SCHEMA_TEXT);
+        properties.add(RECORD_WRITER);
+        properties.add(NUM_RECORDS);
+        properties.add(LOCALE);
+        properties.add(NULLABLE_FIELDS);
+        properties.add(NULL_PERCENTAGE);
+        return properties;
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder()
+                .name(propertyDescriptorName)
+                .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+                .allowableValues(FIELD_TYPES)
```

Comment:
The `Faker` class has an `expression()` method that supports composed references to multiple properties along these lines:

```
#{Name.first_name} #{Name.first_name} #{Name.last_name}
```

The `#{...}` format conflicts with Parameter Names, but we could support a translation approach. It seems like that would provide a great deal of flexibility without locking in particular enumerated types.